File: component_regular_expression/src/main/java/com/cjh/component_regular_expression/MainActivity.java
package com.cjh.component_regular_expression;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

import com.cjh.lib_basissdk.util.FileIOUtils;
import com.cjh.lib_basissdk.util.ResourceUtils;

import java.io.File;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = MainActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //regexBook(readAssets("regular3.txt"));
        regexWidget("regular4.txt");
    }

    private void regexBook(String target) {
        String regex = "《.*?》";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(target);
        while (matcher.find()) {
            //Log.e(TAG, matcher.group());
            System.out.println(matcher.group());
        }
    }

    private String readAssets(String name) {
        return ResourceUtils.readAssets2String(name);
    }

    private static void regexWidget(String target) {
        String regex = "$\\{.*?}";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(target);
        while (matcher.find()) {
            System.out.println(matcher.group());
        }
    }

}

File: component_ipc_service/src/main/java/com/cjh/component_ipc_service/MainActivity.kt
package com.cjh.component_ipc_service

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun goReplaceFragment(view: View) {
        //val intent = Intent(this@MainActivity, ReplaceFragmentActivity::class.java)
        startActivity(intent)
    }

    fun goHideShowFragment(view: View) {
        //val intent = Intent(this@MainActivity, HideShowFragmentActivity::class.java)
        startActivity(intent)
    }

    fun goNormalFragment(view: View) {
        //val intent = Intent(this@MainActivity, NormalFragmentActivity::class.java)
        startActivity(intent)
    }

    fun goLazyLoadFragment(view: View) {
        //val intent = Intent(this@MainActivity, LazyLoadFragmentActivity::class.java)
        startActivity(intent)
    }

    fun goXmlLoadFragment(view: View) {
        //val intent = Intent(this@MainActivity, XmlLoadFragmentActivity::class.java)
        startActivity(intent)
    }

}
